go/git/get.go

```diff
@@ -0,0 +1,123 @@
+	CTime                 time.Time                     `json:"ctime"`
+	MTime                 time.Time                     `json:"mtime"`
+	EncryptedMetadata     string                        `json:"encrypted_metadata"`
+	Nonce                 string                        `json:"nonce"`
+	KeyGeneration         keybase1.PerTeamKeyGeneration `json:"key_generation"`
+	LastModifyingUsername string                        `json:"last_writer_username"`
+	LastModifyingDeviceID keybase1.DeviceID             `json:"last_writer_device_id"`
+}
+
+type ServerResponse struct {
+	Repos []ServerResponseRepo `json:"repos"`
+}
+
+func getMetadataInner(ctx context.Context, g *globals.Context, folder *keybase1.Folder) ([]keybase1.GitRepoResult, error) {
+	// TODO: replace these with real implementations
+	var teamer Teamer
+	var cryptoer Cryptoer
+
+	apiArg := libkb.APIArg{
+		Endpoint:    "git/kbfs/team/get",
+		SessionType: libkb.APISessionTypeREQUIRED,
+		NetContext:  ctx,
+		Args:        libkb.HTTPArgs{
+		// a limit parameter exists, default 100, and we don't currently set it
+		},
+	}
+
+	// The team_id parameter is optional. Add it in if the caller supplied it.
+	if folder != nil {
+		if folder.FolderType != keybase1.FolderType_TEAM {
+			return nil, errors.New("git repo folders must be team folders")
+		}
+		teamIDVis, err := teamer.LookupOrCreate(ctx, *folder)
+		if err != nil {
+			return nil, err
+		}
+		apiArg.Args["team_id"] = libkb.S{Val: string(teamIDVis.TeamID)}
+	}
+	res, err := g.GetAPI().Get(apiArg)
+
+	var resObj ServerResponse
+	err = res.Body.UnmarshalAgain(&resObj)
+	if err != nil {
+		return nil, err
+	}
+
+	var resultList []keybase1.GitRepoResult
+	for _, responseRepo := range resObj.Repos {
+		teamIDVis := keybase1.TeamIDWithVisibility{
+			TeamID: responseRepo.TeamID,
+			// TODO: We're assuming all git repos are for private teams.
+			// Eventually we might add support for public ones?
+			Visibility: keybase1.TLFVisibility_PRIVATE,
+		}
+
+		ciphertext, err := base64.StdEncoding.DecodeString(responseRepo.EncryptedMetadata)
+		if err != nil {
+			return nil, err
+		}
+
+		nonceSlice, err := base64.StdEncoding.DecodeString(responseRepo.Nonce)
+		if err != nil {
+			return nil, err
+		}
+		if len(nonceSlice) != len(keybase1.BoxNonce{}) {
+			return nil, fmt.Errorf("expected a nonce of length %d, found %d", len(keybase1.BoxNonce{}), len(nonceSlice))
+		}
+		var nonce keybase1.BoxNonce
+		copy(nonce[:], nonceSlice)
+
+		msgpackPlaintext, err := cryptoer.Unbox(ctx, ciphertext, nonce, teamIDVis, responseRepo.KeyGeneration)
+		if err != nil {
+			return nil, err
+		}
+
+		var localMetadata keybase1.GitLocalMetadata
+		mh := codec.MsgpackHandle{WriteExt: true}
+		dec := codec.NewDecoderBytes(msgpackPlaintext, &mh)
+		err = dec.Decode(localMetadata)
+		if err != nil {
+			return nil, err
+		}
+
+		resultList = append(resultList, keybase1.GitRepoResult{
+			// TODO: figure out what Folder the team corresponds to
```

Comment:
@mlsteele, what's the right way to go from an implicit team back out to the display name? I see that the `team_implicit` table stores `display_name_hash`, but that doesn't seem to get me there.

Reply:
The team loader can do this.